// wxWindows Patch
// Modified to be compatible with current unreleased library
// 10-24-2001 Robert Cavanaugh
// Description:
// This patch adds a "Help" button event handler and modifies
// the wxWizardEvent class to add the current wxWizardPage
// to the event data

--- Tip/wxWindows/include/wx/wizard.h	Sat Sep 15 04:39:10 2001
+++ New/wxWindows/include/wx/wizard.h	Fri Oct 19 17:30:56 2001
@@ -5,5 +5,7 @@
 // Author:      Vadim Zeitlin (partly based on work by Ron Kuris and Kevin B.
 //              Smith)
-// Modified by:
+// Modified by:	Robert Cavanaugh
+//              Added capability to use .WXR resource files in Wizard pages
+//              Added wxWIZARD_HELP event
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.h,v 1.13 2001/09/15 10:39:10 VS Exp $
@@ -164,5 +175,6 @@
     wxWizardEvent(wxEventType type = wxEVT_NULL,
                   int id = -1,
-                  bool direction = TRUE);
+                  bool direction = TRUE,
+                  wxWizardPage* page = NULL);
 
     // for EVT_WIZARD_PAGE_CHANGING, return TRUE if we're going forward or
@@ -172,6 +184,9 @@
     bool GetDirection() const { return m_direction; }
 
+    wxWizardPage*   GetPage() const { return m_page; }
+
 private:
     bool m_direction;
+    wxWizardPage*    m_page;
 
     DECLARE_DYNAMIC_CLASS(wxWizardEvent)
@@ -186,4 +201,5 @@
     DECLARE_EVENT_TYPE(wxEVT_WIZARD_PAGE_CHANGING, 901)
     DECLARE_EVENT_TYPE(wxEVT_WIZARD_CANCEL, 902)
+    DECLARE_EVENT_TYPE(wxEVT_WIZARD_HELP, 903)
 END_DECLARE_EVENT_TYPES()
 
@@ -200,4 +216,7 @@
 // unless the event handler vetoes the event
 #define EVT_WIZARD_CANCEL(id, fn) DECLARE_EVENT_TABLE_ENTRY(wxEVT_WIZARD_CANCEL, id, -1, (wxObjectEventFunction) (wxEventFunction) (wxWizardEventFunction) & fn, (wxObject *)NULL),
+
+// the user pressed "Help" button 
+#define EVT_WIZARD_HELP(id, fn) DECLARE_EVENT_TABLE_ENTRY(wxEVT_WIZARD_HELP, id, -1, (wxObjectEventFunction) (wxEventFunction) wxWizardEventFunction) & fn, (wxObject *)NULL),
 
 #endif // wxUSE_WIZARDDLG

--- Tip/wxWindows/src/generic/wizard.cpp	Fri Aug 17 11:52:02 2001
+++ New/wxWindows/src/generic/wizard.cpp	Thu Oct 25 10:55:26 2001
@@ -3,5 +3,8 @@
 // Purpose:     generic implementation of wxWizard class
 // Author:      Vadim Zeitlin
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              1) Added capability for wxWizardPage to accept resources
+//              2) Added "Help" button handler stub
+//              3) Fixed ShowPage() bug on displaying bitmaps
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.cpp,v 1.19 2001/08/17 17:52:02 VZ Exp $
@@ -55,8 +58,12 @@
 DEFINE_EVENT_TYPE(wxEVT_WIZARD_PAGE_CHANGING)
 DEFINE_EVENT_TYPE(wxEVT_WIZARD_CANCEL)
+DEFINE_EVENT_TYPE(wxEVT_WIZARD_HELP)
 
 BEGIN_EVENT_TABLE(wxWizard, wxDialog)
     EVT_BUTTON(wxID_CANCEL, wxWizard::OnCancel)
-    EVT_BUTTON(-1, wxWizard::OnBackOrNext)
+    EVT_BUTTON(wxID_BACKWARD, wxWizard::OnBackOrNext)
+    EVT_BUTTON(wxID_FORWARD, wxWizard::OnBackOrNext)
+// 10-20-2001 RAC - Added Help button handler
+    EVT_BUTTON(wxID_HELP, wxWizard::OnHelp)
 END_EVENT_TABLE()

@@ -396,4 +435,20 @@
 }
 
+void wxWizard::OnHelp(wxCommandEvent& WXUNUSED(event))
+{
+    // Added 10-20-2001 Robert Cavanaugh
+    // this function probably can never be called when we don't have an active
+    // page, but a small extra check won't hurt
+    if(m_page != NULL)
+    {
+        // Create and send the help event to the specific page handler
+        // event data contains the active page so that context-sensitive
+        // help is possible
+        wxWizardEvent NewEvent(wxEVT_WIZARD_HELP, GetId(), TRUE, m_page);
+        (void)m_page->GetEventHandler()->ProcessEvent(NewEvent);
+    }
+}
+
+
 // ----------------------------------------------------------------------------
 // our public interface
@@ -415,8 +470,11 @@
 // ----------------------------------------------------------------------------
 
-wxWizardEvent::wxWizardEvent(wxEventType type, int id, bool direction)
+wxWizardEvent::wxWizardEvent(wxEventType type, int id, bool direction, wxWizardPage* page)
              : wxNotifyEvent(type, id)
 {
+    // Modified 10-20-2001 Robert Cavanaugh
+    // add the active page to the event data
     m_direction = direction;
+    m_page = page;
 }




 

 	  	 
